src/Components/CreateImageWizardV2/steps/TargetEnvironment/Aws/AwsSourcesSelect.tsx

```diff
@@ -31,7 +31,7 @@ export const AwsSourcesSelect = () => {
     _event: React.MouseEvent<Element, MouseEvent>,
     value: string
   ) => {
-    const source = sources?.find((source) => source.name === value);
+    const source = sources?.find((source) => source?.name === value);
```

Comment:
Do we need the optional chaining here?  In the API schema name is already optional and it can be undefined.

```typescript
(parameter) source: {
    id?: string | undefined;
    name?: string | undefined;
    source_type_id?: string | undefined;
    uid?: string | undefined;
}
```

Reply:
Unresolving - tried to remove the optional chaining and I can't see any errors or problems. I'd suggest removing it:
```suggestion
    const source = sources?.find((source) => source.name === value);
```